### ISSUE TITLE
any ssh or scp TAB leads to "complete: maximum recursion depth reached" infloop

### ISSUE THREAD

**Issue by ilohmar** [fish-shell/fish-shell#7389]
Please tell us which fish version you are using by executing the following:
fish, version 3.1.2

Please tell us which operating system and terminal you are using. The output of `uname -a` and `echo $TERM` may be helpful in this regard although other commands might be relevant in your specific situation.
Linux t430il 5.8.0-2-amd64 #1 SMP Debian 5.8.10-1 (2020-09-19) x86_64 GNU/Linux
st-256color

The issue does not occur in the no-customizations setup, where I do not get any TAB completions at all.  I have an
```
alias ssh="env TERM=screen command ssh"
```
but removing it (functions -e ssh) does not change the situation.  I also tried to mv my ssh config out of the way, also without any effect.  That's when I was a bit stumped --- what could go wrong here?

**Comment by faho** [comment]
See also #6899

**Comment by ilohmar** [comment]
That's why I checked if removing the alias function makes a difference -- it did not.

**Comment by faho** [comment]
Please remove it *in a new shell*

**Comment by faho** [comment]
As in start a shell that never has it. Remove it from the config.

**Comment by ilohmar** [comment]
That works, indeed, thank you!  And now I have two Q.s :)

1) What is "left" of the alias function for ssh when I do `functions -e ssh` (which was meant to get rid of it)?

2) What is wrong with `alias ssh="env TERM=screen command ssh"` that lets the completion fail?

If this is doc.d, I would appreciate pointers!

**Comment by faho** [comment]
Alias defines completions. In this case it tells fish to complete your alias like the command it calls - env. Only the env completions say to complete like the subcommand, ssh. Which is also the name of your alias. Which results in an infinite loop. 

The completions remain defined even if the function is erased. 

What you can do is either

-make it an actual function directly

```fish
function ssh
    TERM =xterm-256color command ssh $argv
end
```

(no need for the env in fish 3.1) 

or to call it something else.

**Comment by ilohmar** [comment]
> Alias defines completions.

I did not consider that, but it makes a lot of sense.  Thanks for the explanation.

**Comment by faho** [comment]
Reopening this one because it shouldn't fail like that.

**Comment by krobelus** [comment]
Okay, the completion repeated ad infinitum because the `env` completions called subcommand completions multiple times, instead of just once.
Fixed that, and modified `alias` to not define those recursive completions in the first place.

`alias ssh "TERM=screen ssh"` already worked on master before my fix, but not on 3.1.2.
This seems to be a common pattern in other shells.

> alias ssh="env TERM=screen command ssh"

A shell gotcha: `command` is a shell builtin, but `env` is an external program. Hence `command` will not work, and is actually not needed since `env` can only see external commands anyway.

**Comment by ilohmar** [comment]
> A shell gotcha: command is a shell builtin, but env is an external program.
> Hence command will not work, and is actually not needed since env can only see external commands anyway.

Thanks, realized that when writing a temporary workaround function.  Thanks for your help and for the fix!
